core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -122,6 +126,14 @@ class SparkHadoopUtil extends Logging {
   }
 
   /**
+   * Schedule a login from the keytab and principal set using the --principal and --keytab
+   * arguments to spark-submit. This login happens only when the credentials of the current user
+   * are about to expire. This method reads SPARK_PRINCIPAL and SPARK_KEYTAB from the environment
+   * to do the login. This method is a no-op in non-YARN mode.
+   */
+  private[spark] def scheduleLoginFromKeytab(): Unit = {}
```

Comment:
The reason is that this is used from CoarseGrainedSchedulerBackend. Either we have this here or have a match-case in that class. I'd rather have this method as the code there becomes cleaner.